internal/host/validator.go

```diff
@@ -1537,52 +1545,56 @@ func (v *validator) isReleaseDomainResolvedCorrectly(c *validationContext) (Vali
 
-	if wildcardDomainResolution == nil {
+	if domainResolution == nil {
 		return ValidationError, "Internal error - DNS wildcard check cannot be performed because the DNS validation result doesn't contain the wildcard domain"
 	}
 
-	if len(wildcardDomainResolution.IPV4Addresses) == 0 && len(wildcardDomainResolution.IPV6Addresses) == 0 {
+	if len(domainResolution.IPV4Addresses) == 0 && len(domainResolution.IPV6Addresses) == 0 {
 		return ValidationSuccess, "DNS wildcard check was successful"
 	}
 
 	// Compile a list of IP addresses that the bad wildcard domain resolves to
 	// so they can be included in the message. This might help users understand
 	// where the bad wildcard record is coming from.
 	addressStrings := []string{}
-	for _, ipv4Address := range wildcardDomainResolution.IPV4Addresses {
+	for _, ipv4Address := range domainResolution.IPV4Addresses {
 		addressStrings = append(addressStrings, string(ipv4Address))
 	}
-	for _, ipv6Address := range wildcardDomainResolution.IPV6Addresses {
+	for _, ipv6Address := range domainResolution.IPV6Addresses {
 		addressStrings = append(addressStrings, string(ipv6Address))
 	}
 
-	return ValidationFailure, fmt.Sprintf("DNS wildcard configuration was detected for domain *.%s.%s - the installation will not be able to complete while this record exists. Please remove it to proceed. The domain resolves to addresses %s",
-		c.cluster.Name, c.cluster.BaseDNSDomain, strings.Join(addressStrings, ", "))
+	return ValidationFailure, fmt.Sprintf(messageFormat, c.cluster.Name, c.cluster.BaseDNSDomain, strings.Join(addressStrings, ", "))
+}
+
+func (v *validator) isDNSWildcardNotConfigured(c *validationContext) (ValidationStatus, string) {
+	if c.infraEnv != nil {
+		return ValidationSuccessSuppressOutput, ""
+	}
+	if hostutil.IsDay2Host(c.host) {
+		return ValidationSuccess, "DNS wildcard check is not required for day2"
+	}
+	if c.host.DomainNameResolutions == "" {
+		return ValidationPending, "DNS wildcard check cannot be performed yet because the host has not yet performed DNS resolution"
+	}
+	var response *models.DomainResolutionResponse
+	if err := json.Unmarshal([]byte(c.host.DomainNameResolutions), &response); err != nil {
+		return ValidationError, "Error while parsing DNS resolution response"
+	}
+
+	status, message := checkUnexpectedDomain(c, response, dottedDomainNameToResolve(c, constants.DNSWildcardFalseDomainName),
+		"DNS wildcard configuration was detected for domain *.%s.%s - the installation will not be able to complete while this record exists. Please remove it to proceed. The domain resolves to addresses %s")
+	if status == ValidationSuccess {
+		status, message = checkUnexpectedDomain(c, response, domainNameToResolve(c, constants.DNSWildcardFalseDomainName),
+			"Unexpected domain name resolution was detected for sub-domain .%s.%s - the installation will not be able to complete while this resolution exists. Please remove it to proceed. The domain name resolves to addresses %s")
```

Comment:
How about something like

```suggestion
			"Unexpected domain name resolution was detected for sub-domain *.%[1]s.%[2]s despite no resolution for a similar domain with a train dot: *.%[1]s.%[2]s. - this is usually a sign of DHCP-provided domain-search configuration. The installation will not be able to complete with this configuration in place. Please remove it to proceed. The domain name resolves to addresses %[3]s")
```